### ISSUE TITLE
Tags should be required when creating a Collective

### ISSUE THREAD

**Issue by Memo-Es** [opencollective/opencollective#6077]
## Context

This is a follow-up of https://github.com/opencollective/opencollective/issues/4764. We are investing a considerable amount of resources in creating new features such as the new discovery [(a.k.a. Collective weather report)](https://docs.google.com/document/d/1Eh_vbV7sK9FQ5HmS0P9Zqz3LReOgTdMf8BJJUcHGpI8/edit?usp=sharing), the Funders Contributors' dashboard, and the search-related targets like #3892 #3782 https://github.com/opencollective/opencollective/issues/5793 https://github.com/opencollective/opencollective/issues/5809 .
It is also necessary for fiscal hosts admin teams to have a better understandment of the communities they host and a cleaner visualization in categories.

## Goal

 - [ ] Make the tags required in the Collective creation form https://opencollective.com/create
 - [ ] Update the form considering https://github.com/opencollective/opencollective/issues/5628

@Betree @kaylarep

**Comment by Betree** [comment]
No real blocker on the engineering side, but we'll need a design for the error state of this field when it's required (I don't think we have a similar case of required tags in the UI).

![image](https://user-images.githubusercontent.com/1556356/198219667-5a5f688e-dd81-4a42-ab33-d7f180cedf6f.png)

From a personal point of view, I don't really see the justification for making it required: it's going to increase the complexity of creating a collective, but are tags that important? 

- For search-related issues, the description (which is already required) and the long description are already included in the searched fields. These fields provide more information than tags, and better context for host admins to understand their communities.
- For the discovery project, what kind of tags do we expect to see? I've seen the location mentioned a few times, but tags are not the proper way to input one's location; we need more structured data if we want to make the search effective.

To me, tags are part of the onboarding (aka. make your profile better) flow, not the create flow. They're currently not requested in the onboarding modal.

**If we do want** to enforce tags in the Create Collective flow, then I would suggest we first define what kind of tags we want people to set so that we kind provide meaningful suggestions; there's otherwise a risk of ending up with quantity without quality.

**Comment by kaylarep** [comment]
Thanks for opening this @Memo-Es 

Yes, @Betree I believe tags to be a very important tool. I've raised this topic many times, it's come up again because of Caroline's project for a Funder's Dashboard. 

Right now, we have so many groups using the platform, and we really have no idea what we have. In my mind, tagging would be the most effective way for:
- groups with common interests to find each other (community-building)
- donors to find groups to donate to (especially if they can only donate to specific sector)

Ideally, yes it would be in the onboarding/make your profile better flow, but since we are not prioritizing this right now, this is a temporary fix to surface groups better (and especially for the discovery project, which has a deadline coming). As the platform is right now, many groups do not prioritize making their profile's better, and may not even understand the benefit of doing it.

I have been asked to go through and manually tag all of OCF's groups. I have already done this 1 time before, and it was incredibly labor-intensive with a big margin for error.  I have been asked to do it again, to catch up with all the groups who have joined since I last completed the task, but I do not think it is a wise use of time to be manually doing this, also yes, there is room for using the wrong term to tag/identify a group, and possibly even mislabel and offend them.

If I will have to manually do it again, I think it should be the last time I do it.  Which means this issue needs to be prioritized so that groups should be tagging themselves (which is definitely preferred).

I would like groups to be "required" to select a few tags, but yes also have the option to skip it if they'd like. (As described here: https://github.com/opencollective/opencollective/issues/4764 )

___

> If we do want to enforce tags in the Create Collective flow, then I would suggest we first define what kind of tags we want people to set so that we can provide meaningful suggestions; there's otherwise a risk of ending up with quantity without quality. 
 
+1
My recommendations for this are:
- https://github.com/opencollective/opencollective/issues/5523 (this way, naturally occurring terms that groups use to self-identify will come to the surface)
- in addition to other Tag improvements like:
    - https://github.com/opencollective/opencollective/issues/5522
    - https://github.com/opencollective/opencollective/issues/5521  
    - there are more, but I'll leave them out for now

___
> For search-related issues, the description (which is already required) and the long description are already included in the searched fields. These fields provide more information than tags, and better context for host admins to understand their communities.

There is not currently (or probably ever) a completely standardized approach/wordage that groups will use to describe themselves, but they will certainly have a lot of overlap. To me, this is the idea behind, and benefit of, tagging.  e.g. If I describe my group in California as a "community refrigerator" and a group in New York describes themselves as a "free fridge", how would a donor know what to search for to find all these groups, which are in fact doing the same thing.  Ideally (if I understand the mechanism correctly), if we can implement - https://github.com/opencollective/opencollective/issues/5523 ; then when I type in `community refrigerator` as a tag for my group, a suggestion would immediately appear to also select the tag `free fridge`, and by having both tags on my profile, I am widening my audience (also likely to attract donors better/easier).  Auto-populating suggestions would be an incredibly powerful tool in connecting like-minded groups; also in helping donors easily locate the groups they want to support.

**Comment by Memo-Es** [comment]
I see this as a medium project, regardless if it is on the form or elsewhere (BTW, the form is a main piece of the onboarding flow). I lean towards having at least one tag on ALL Collectives for the organization and quality of our Collective categorization.
A couple of preliminary points that will need clarification:
 - Automatic tags? (Collectives under OSC should be tagged open-source automatically?)
 - What are the cases of a Collective not wanting tags? Collectives that do not want to be found?
 - Define the experience of **Suggesting tags to Collectives**, related with https://github.com/opencollective/opencollective/issues/5523 but it may include some other stuff as well

I agree with @kaylarep that manually tagging Collectives is a poor way to invest our time. Happy to see this project prioritized for this next cycle.

**Comment by Betree** [comment]
Thanks for providing more context, I am now convinced that tags can be a good tool for the described problems. Also good that we already have issues about improving the suggestions, we should definitely pursue both goals (more tags and better tags) in parallel!

**Comment by kaylarep** [comment]
@Betree Does this mean this will get prioritized? 

I'm still receiving applications without tags, and I really don't want to go thru the manual tagging project again until this will be implemented.


@Memo-Es My responses to your questions:
> Automatic tags? (Collectives under OSC should be tagged open-source automatically?)
- I do not think we should do auto-tagging.There's too much room for error. (e.g. a group submits an application to OSC, receives an `open source` tag, but they are not actually an open-source project or are eligible for hosting thru OSC, and then they just have an incorrect tag attributed to their collective.

> What are the cases of a Collective not wanting tags? Collectives that do not want to be found?
- Yes, this could be one example. Or they intentionally are keeping their Collective page very sparse and are instead investing their energy into their own website.  Many of our groups are serving sensitive populations, and being "discoverable" is not their top priority.

> Define the experience of Suggesting tags to Collectives, related with https://github.com/opencollective/opencollective/issues/5523 but it may include some other stuff as well
- this is the general idea: https://www.loom.com/share/1c161af00e624d09a55fb7fd620e4287

**Comment by kaylarep** [comment]
also, (and please disregard if I'm speaking completely out of my depth here) if we are able to decide how the auto-suggestions are formulated, I think I would prefer a community-based approach. e.g. I select `arts` and then "Other groups also used: `arts & culture` `painting` `theatre` `youth` " would appear- because that's what other groups who used `arts` have used.

as opposed to a more computational approach of AI/algorithm guessing what categories are similar

**Comment by kaylarep** [comment]
something that looks like this? 
![image](https://user-images.githubusercontent.com/64920686/200405493-4947e7d3-86d2-4e1e-ab69-b9e7d2ef4225.png)
